### PR TITLE
GH-2306 more robust check for external bindings in FedXConnection

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #2306  

Briefly describe the changes proposed in this PR:

- in FedXConnection, when checking if supplied bindings contain anything other than the "special" FedX bindings, do not assume we know the exact number of supplied FedX bindings, as it can vary.
- in SailUpdateExecutor, when merging result bindings with preset bindings, ensure a binding is not null before merging.

---- 
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [x] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

**Comment by jeenbroekstra** [comment]
We get a nullpointerexception "in the wild" when using this in a strange corner case. I have yet to succeed in constructing a simple test case, but regardless: since getBinding is allowed to return `null` and addBinding is expected to throw an error if null is supplied, this check should be there.

**Comment by jeenbroekstra** [comment]
The actual root cause of the issue was a bug in FedXConnection, which explains why it was  not reproducible on other Sails: it failed to inject an external binding. Fix added.

File: tools/federation/src/main/java/org/eclipse/rdf4j/federated/FedXConnection.java [hunk 1]
```diff
@@ -138,10 +138,8 @@ public FedXConnection(FedX federation, FederationContext federationContext)
 
 		try {
 			// make sure to apply any external bindings
-			// Note: we use a conditional check for performance reasons.
-			// Further note that we subtract -1 from the size as the baseUri is optional
 			BindingSet queryBindings = EmptyBindingSet.getInstance();
-			if (bindings.size() > FedXRepositoryConnection.FEDX_BINDINGS.size() - 1) {
+			if (!FedXRepositoryConnection.FEDX_BINDINGS.containsAll(bindings.getBindingNames())) {
```
Comment (jeenbroekstra) [inline tools/federation/src/main/java/org/eclipse/rdf4j/federated/FedXConnection.java:142]: Fwiw I believe this is still a check worth doing as in most cases it will skip creating a new MapBindingSet, and instead just pass in the `EmptyBindingSet` singleton - which other parts of the engine have some optimizations for.

Comment (aschwarte10) [reply tools/federation/src/main/java/org/eclipse/rdf4j/federated/FedXConnection.java:142]: Good solution and way more robust. Should do the job as expected
